tools/lint.py

```diff
@@ -71,7 +71,7 @@ def run_ruff(files, fix):
         return 0, ""
     args = ['--fix'] if fix else []
     res = subprocess.run(
-        ['ruff', f'--config={CONFIG}'] + args + files,
+        ['ruff', f'--config={CONFIG}'] + args + list(files),
```

Comment:
Isn't files already guaranteed to be a list?

Reply:
Not if sets are used below -- `files` here is the result of a set difference